apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/CorrelationContext.java

```diff
@@ -78,6 +78,11 @@ public CorrelationContext() {
         // already contain key
         if (data.containsKey(key)) {
             final String previousValue = data.put(key, value);
+
+            // update tag if need
+            if (AUTO_TAG_KEYS.contains(key) && ContextManager.isActive()) {
+                ContextManager.activeSpan().tag(new StringTag(-1, key, true), value);
```

Comment:
Why -1 as key?

Reply:
I am not sure about your meaning of update. Transitionally, we don't encourage tag update, if it is really needed.

Reply:
`-1` is the tag `id`. It uses to compare with the same tag. In the `StringTag` constructor, the id is `-1` default.

Reply:
My point is, why override? Overrideable tags are limited. What is the use case?

Reply:
> I am not sure about your meaning of update. Transitionally, we don't encourage tag update, if it is really needed.

I think the correlation tag value is overridden, the same segment tag should be synchronized.

Reply:
Do you mean same tag key? If so, no, it should not override. Most keys in current plugins are not overriding.

Reply:
I see. This PR will be closed.

Reply:
OK, I see what you are missing. CorrelationContext is not relating to or connecting with the span of segment. Instead, CorrelationContext is totally separated with tracing context, but leveraging the context propagation mechanism for assisting the users.
The codes here are for a very specific cases, which mean, there is a context propagated from remote peer, and need to inject into new context(mostly because there is an entry span), then the tags will be set. That is all. No further interaction.